flask_securest/authentication_providers/password.py

```diff
@@ -36,29 +39,45 @@ class PasswordAuthenticator(AbstractAuthenticationProvider):
 
     def __init__(self, password_hash=DEFAULT_PASSWORD_HASH):
         self.crypt_ctx = _get_crypt_context(password_hash)
+        self.request_user_id = None
+        self.request_password = None
 
-    def authenticate(self, auth_info, userstore):
-        if not auth_info.user_id or not auth_info.password:
-            raise Exception('username or password not found on request')
+    def _retrieve_request_credentials(self):
+        auth_header = request.headers.get(AUTH_HEADER_NAME)
+        if not auth_header:
+            raise RuntimeError('Request authentication header "{0}" is empty '
+                               'or missing'.format(AUTH_HEADER_NAME))
 
-        user_id = auth_info.user_id
-        user = userstore.get_user(user_id)
+        auth_header = auth_header.replace(BASIC_AUTH_PREFIX + ' ', '', 1)
+        try:
+            from itsdangerous import base64_decode
+            api_key = base64_decode(auth_header)
+            # TODO parse better, with checks and all, this is shaky
+        except TypeError:
+            pass
+        else:
+            api_key_parts = api_key.split(':')
+            self.request_user_id = api_key_parts[0]
```

Comment:
it is also worth validating the `len(api_key_parts) == 2` otherwise you will never reach line 62 and this call will fail in line 61 with `IndexError`